### PR TITLE
optimize torch performance

### PR DESCRIPTION
**Pull request by haifeng-jin**

### CODE REVIEW

File: keras_core/backend/torch/core.py [hunk 1]
```diff
@@ -170,7 +160,10 @@ def cast(x, dtype):
     if isinstance(x, KerasVariable):
         x = x.value
     if is_tensor(x):
-        return x.to(dtype)
+        if x.dtype == dtype:
+            return x
+        else:
+            return x.to(dtype)
```
Comment (old-school-kid) [inline keras_core/backend/torch/core.py:166]: Why are we putting an overhead of if-else here?

File: keras_core/backend/torch/core.py [hunk 2]
```diff
@@ -114,32 +106,30 @@ def __eq__(self, other):
...
 
 def convert_to_tensor(x, dtype=None):
-    dtype = to_torch_dtype(dtype or getattr(x, "dtype", None))
-    device = get_device()
-    if isinstance(x, int):
-        dtype = torch.int32
-    if isinstance(x, float):
-        dtype = torch.float32
+    if is_tensor(x):
+        return x
```
Comment (old-school-kid) [inline keras_core/backend/torch/core.py:110]: Will have to check dtype and convert to the said dtype if dtype is passed. Tests ae failing when we pass integer arguments with dtype boolean

Comment (fchollet) [reply keras_core/backend/torch/core.py:110]: Yes, let's add the check and conversion here.

File: keras_core/backend/torch/core.py (hunk 1 above)
Comment (haifeng-jin) [reply keras_core/backend/torch/core.py:166]: It turns out `x.to` has more overhead. This would save some time by only call it when necessary.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**